doc/nrf/device_guides/working_with_nrf/nrf53/qspi_xip_guide.rst

```diff
@@ -0,0 +1,321 @@
+.. _qspi_xip:
+
+Configuring external XIP
+########################
+
+.. contents::
+   :local:
+   :depth: 2
+
+This section describes the external XIP support for the nRF5340 SoC.
+The nRF5340 SoC is equipped with QSPI memory interface which is capable of exposing QSPI flash as memory for CPU used to fetch and execute the program code.
+The QSPI is available for the application core.
+Therefore, it is possible to relocate a part of the application's code to an external memory.
+NCS supports dividing an application into an internal and external part, along with the MCUboot support.
+
+To configure a project for QSPI XIP, you need to perform the following steps specific to the ext-xip feature:
+
+ * Provide QSPI flash memory DTS.
+ * Proviode a dedicated static partition manager configuration file with proper internal and external partitioning.
+ * Provide a linker file that describes the memory block of the QSPI XIP code location.
+ * Describe code relocation in the project's CMake file.
+
+Additionally, regular project setup elements are mandatory, for example, an application for the application core and a networking stack for the networking core.
+You will need to further configure those elements.
+
+Enabling configuration options
+******************************
+
+Application configuration must support an external XIP and image splitting.
+
+Enable the following options:
+
+* :kconfig:option:`CONFIG_CODE_DATA_RELOCATION`
+* :kconfig:option:`CONFIG_HAVE_CUSTOM_LINKER_SCRIPT`
+* :kconfig:option:`CONFIG_BUILD_NO_GAP_FILL`
+* :kconfig:option:`CONFIG_XIP`
+* :kconfig:option:`CONFIG_NORDIC_QSPI_NOR_XIP`
+* :kconfig:option:`CONFIG_XIP_SPLIT_IMAGE`
+* :kconfig:option:`CONFIG_BOOTLOADER_MCUBOOT` - This option enables the MCUboot build and support for the application.
+
+Additionally set the following options:
+
+* :kconfig:option:`CONFIG_CUSTOM_LINKER_SCRIPT` to ``"<linker_file_for_relocation>"``
+* :kconfig:option:`CONFIG_UPDATEABLE_IMAGE_NUMBER` to ``3``
+* :kconfig:option:`CONFIG_FLASH_INIT_PRIORITY` to ``40`` - It is required to ensure the QSPI device initialization priority, which makes the external XIP code accessible before it is executed.
+  If any initialization code is expected to be run from the QSPI XIP, then its initialization priority value must be lower than the QSPI device initialization priority.
+
+Setting up QSPI flash
+*********************
+
+The QSPI flash DTS must be available for the application and the MCUboot.
```

Comment:
```suggestion
The QSPI flash DTS file must be available for the application and the MCUboot.
```
Not sure if correct? "DTS variable" otherwise?

Reply:
Yes, you're right, I must've added this unconsciously